### PR TITLE
Questions for design section

### PR DESCRIPTION
**Pull request by sacha-l**
This creates new content for:
- The [design landing page](https://github.com/substrate-developer-hub/docs-sandbox/blob/f68147d045202bd743de784a53fa5a9504034eea/source/docs/main-docs/05-design/index.md)
- The [implementation details page](https://github.com/substrate-developer-hub/docs-sandbox/blob/f68147d045202bd743de784a53fa5a9504034eea/source/docs/main-docs/05-design/implementation-details.md)
- The [economic models page](https://github.com/substrate-developer-hub/docs-sandbox/blob/f68147d045202bd743de784a53fa5a9504034eea/source/docs/main-docs/05-design/economic-models.md)

### CODE REVIEW

File: source/docs/main-docs/05-design/economic-models.md [hunk 1]
```diff
@@ -2,6 +2,13 @@ Section: Design
 Sub-section: Economic models
 Type: reference + conceptual
 
+Questions: 
+
+Scenario: I'm building a parachain and I'm looking to ensure my chain is economically secure. 
+
+- What sort of economic models come "out of the box" with Substrate? What safe guards?
```
Comment (shawntabrizi) [inline source/docs/main-docs/05-design/economic-models.md:9]: We have a bunch of pallets which provide some "safe economic systems" like staking, assets, uniques, etc...

In terms of economics systems provided by FRAME itself, just the use of tokens really. So slashing tokens when they do something bad, minting tokens when they do something good, reserving tokens to incentivize people to unreserve those tokens.

File: source/docs/main-docs/05-design/economic-models.md [hunk 2]
```diff
@@ -2,6 +2,13 @@ Section: Design
...
+- Can I assess my economic design against a threat model common to other Substrate chains?
```
Comment (shawntabrizi) [inline source/docs/main-docs/05-design/economic-models.md:10]: I really dunno how to respond to this question. Seems very broad.

File: source/docs/main-docs/05-design/index.md [hunk 3]
```diff
@@ -4,20 +4,28 @@ Type: conceptual
...
 <!-- _Notes to docs team: This artice aims to describe the concepts around developing Substrate runtimes._
-_This should be the entry point for anyone developing with substrate, whereby this content conceptually describes the ways one can create application specific business logic for a substrate chain. At a high level, this includes designing new pallets or smart contracts or using both. Runtime development ultimately encompasses the two._ -->
+_This should be the entry point for anyone developing with substrate, whereby this content conceptually describes the ways one can create application specific business logic for a substrate chain._ -->
 # Designing your runtime
+
+TODO: Improve this section's write-up based on these guiding questions.
+
+Scenario: I want to understand what components I need to consider when designing my runtime. I'd then like to prototype my idea, or jump right into using existing pallets and creating my own. This is the lay of the land before I get there.
+
+- At a HL, what does it mean to design a runtime? What are the different components I should focus on?
```
Comment (shawntabrizi) [inline source/docs/main-docs/05-design/index.md:14]: Designing a runtime is all about making your blockchain useful to users. All the non-runtime parts of a blockchain are making sure that things stay in consensus, connecting to peers, etc...

The runtime is the actual logic and application of the blockchain, and really designing the runtime is just like designing an application.

File: source/docs/main-docs/05-design/index.md [hunk 4]
```diff
@@ -4,20 +4,28 @@ Type: conceptual
...
+- For a team getting started, what's a good way to prototype a chain's logic? What factors would they need to consider?
```
Comment (shawntabrizi) [inline source/docs/main-docs/05-design/index.md:15]: I guess they could prototype things as a smart contract first, but really best thing is just to start writing pallets for a solo chain and writing tests.

Ultimately they need to consider scaling and attack vectors to the design of their pallets.

File: source/docs/main-docs/05-design/index.md [hunk 5]
```diff
@@ -4,20 +4,28 @@ Type: conceptual
...
+- How can I determine whether my prototype is a good canadidate to become a chain in the first place? Where should I go with my prototype if not?
```
Comment (shawntabrizi) [inline source/docs/main-docs/05-design/index.md:16]: We cannot tell people whether their idea will be a good chain. its like asking if something will be a good business...

File: source/docs/main-docs/05-design/index.md [hunk 6]
```diff
@@ -4,20 +4,28 @@ Type: conceptual
...
 # Designing your runtime
+
+TODO: Improve this section's write-up based on these guiding questions.
+
+Scenario: I want to understand what components I need to consider when designing my runtime. I'd then like to prototype my idea, or jump right into using existing pallets and creating my own. This is the lay of the land before I get there.
+
+- At a HL, what does it mean to design a runtime? What are the different components I should focus on?
+- For a team getting started, what's a good way to prototype a chain's logic? What factors would they need to consider?
+- How can I determine whether my prototype is a good canadidate to become a chain in the first place? Where should I go with my prototype if not? 
+- Now that I know I want to design a parachain, what's my next step?
```
Comment (shawntabrizi) [inline source/docs/main-docs/05-design/index.md:17]: You should begin by building a solo chain providing the functionality you want, then integrating cumulus into that chain, then connecting it to one of the polkadot test networks, and eventually working to get a parachain slot on the real deal.

File: source/docs/main-docs/05-design/index.md [hunk 7]
```diff
@@ -4,20 +4,28 @@ Type: conceptual
...
+- Can you explain how having full control of the blockchain relates to how I can design my chain?
```
Comment (shawntabrizi) [inline source/docs/main-docs/05-design/index.md:18]: I dont really know how to respond to this. Obviously if you can do more things with substrate, you can design more kinds of different systems. For example applications which trigger some logic at the beginning or end of each block

**Comment by lsgunnlsgunn** [comment]
Some things that come to mind for me:
Identify your goals
What are you building the blockchain for, distributing nfts? Defi app? Serving the needs of a particular community? Maybe some of the projects can give us additional insights into what projects are possible. 
What data types do you need?
What data can you store and how can you minimize ui for performance and efficiency. 
What trade offs are you willing to make for security, for ease of development 
Consensus model
How will you protect the network from malicious activity (transaction fees, staking)? Will the project e economically sustainable? How will you implement on chain governance? This is where I see "planning what to build" and when you have answers most of these questions you can take a staged approach of solochain etc with a Proof of concept or prototype. 
(I'm not sure if this is helpful, but it's where my mind was at.)

**Comment by shawntabrizi** [comment]
In response to @lsgunnlsgunn, I feel in the documentation team we should focus more on strict facts and direction information rather than VERY opinionated design tips and project planning which seems to be triggered by these topics.

For example, rather than writing documentation which helps prompt the user to ask "What data types do you need?" we should simply list all of our data types and what they provide.

Rather than "How will you protect the network from malicious activity (transaction fees, staking)?", lets just simply state the models we have used, and let them figure out their own thing. 

Maybe I am reading the comment wrong, but I think trying to answer these questions in general will lead to really fluffy writing.

File: source/docs/main-docs/05-design/pallet-design.md [hunk 8]
```diff
@@ -1,4 +1,12 @@
- 
+Questions for this page:
```
Comment (sacha-l) [inline source/docs/main-docs/05-design/pallet-design.md:1]: Something to include here: pallets that manipulate data, like blog posts. Best practices for storing only the hash on chain.

**Comment by lisa-parity** [comment]
> simply state the models we have used, and let them figure out their own thing.
Yes, I agree. I'm thinking more about giving people ideas about what's possible/available/been done but not recommending any particular strategy -- leaving it open-ended (here are models we've seen, you can choose one of them and tailor it to your project or develop something completely new). Unless there's specific guidance (pros and cons for each approach) to offer, we just need to let devs know that they have some decisions to make.
Maybe something about game theory belongs in this section too (possibly, if this as a "sketching-out" phase, it might fit?).

**Comment by sacha-l** [comment]
This is pretty much WIP for the design section. I wrote content for the landing page about choosing between pallets or smart contracts (in the [design landing page](https://github.com/substrate-developer-hub/docs-sandbox/blob/f68147d045202bd743de784a53fa5a9504034eea/source/docs/main-docs/05-design/index.md)). And add new content to the Economic models page. Should be good to port out of the sandbox, these articles will all need shaping up once we have a the more fundamental content down. 
cc: @lisa-parity

**Pull request merged**